src/shogun/regression/KRRNystrom.cpp

```diff
@@ -0,0 +1,139 @@
+
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (W) 2016 Fredrik Hallgren
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#include <limits>
+#include <shogun/regression/KRRNystrom.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/mathematics/Math.h>
+
+using namespace shogun;
+using namespace Eigen;
+
+CKRRNystrom::CKRRNystrom() : CKernelRidgeRegression()
+{
+	init();
+}
+
+CKRRNystrom::CKRRNystrom(float64_t tau, int32_t m, CKernel* k, CLabels* lab)
+: CKernelRidgeRegression(tau, k, lab)
+{
+	init();
+
+	m_num_rkhs_basis=m;
+}
+
+void CKRRNystrom::init()
+{
+	m_num_rkhs_basis=1000;
+}
+
+SGVector<int32_t> CKRRNystrom::subsample_indices()
+{
+	int32_t n=kernel->get_num_vec_lhs();
+	SGVector<int32_t> temp(n);
+	temp.range_fill();
+	CMath::permute(temp);
+	SGVector<int32_t> col(m_num_rkhs_basis);
+	for (auto i=0; i<m_num_rkhs_basis; ++i)
+		col[i]=temp[i];
+	CMath::qsort(col.vector, m_num_rkhs_basis);
+
+	return col;
+}
+
+bool CKRRNystrom::solve_krr_system()
+{
+	int32_t n=kernel->get_num_vec_lhs();
+
+	REQUIRE(m_num_rkhs_basis <= n, "Number of sampled rows/columns must be less than number of data\n");
```

Comment:
Should really print out both m_num_rkhs_basis and n here. Otherwise users might be confused.
Also, the check should be done in the constructor and setter rather than here.